pyneuroml/plot/PlotSpikes.py

```diff
@@ -187,139 +191,92 @@ def read_sonata_spikes_hdf5_file(file_name: str):
+                        - "ids" (List[int]): List of cell IDs corresponding to each spike time.
+    :type spike_data: List[Dict[str, Union[List[float], List[int]]]]
+    :param show_plots_already: Whether to show the plots immediately after they are generated. Defaults to True.
+    :type show_plots_already: bool
+    :param save_spike_plot_to: Path to save the spike plot to. If `None`, the plot will not be saved. Defaults to `None`.
+    :type save_spike_plot_to: Optional[str]
+    :param rates: Whether to plot rates in addition to spike times. Defaults to False.
+    :type rates: bool
+    :param rate_window: Window size for rate calculation in ms. Defaults to 50.
+    :type rate_window: int
+    :param rate_bins: Number of bins for rate histogram. Defaults to 500.
+    :type rate_bins: int
+    :return: None
+    :rtype: None
+    """
 
     xs = []
     ys = []
     labels = []
     markers = []
     linestyles = []
 
-    offset_id = 0
-
     max_time = 0
     max_id = 0
+    min_id = float("inf")
     unique_ids = []
+    for data in spike_data:
+        unique_ids.extend(data["ids"])
+    unique_ids = list(set(unique_ids))
     times = OrderedDict()
     ids_in_file = OrderedDict()
 
-    if a.format == "sonata" or a.format == "s":
-        for file_name in a.spiketime_files:
-            ids_times_pops = read_sonata_spikes_hdf5_file(file_name)
+    current_offset = offset
 
-            for pop in ids_times_pops:
-                ids_times = ids_times_pops[pop]
+    for data in spike_data:
+        x = [t for t in data["times"]]
+        y = [id_shifted + current_offset for id_shifted in data["ids"]]
 
-                x = []
-                y = []
-                max_id_here = 0
-
-                name = file_name.split(" / ")[-1]
-                if name.endswith("_spikes.h5"):
-                    name = name[:-10]
-                elif name.endswith(".h5"):
-                    name = name[:-3]
-                times[name] = []
-                ids_in_file[name] = []
-
-                for id in ids_times:
-                    for t in ids_times[id]:
-                        id_shifted = offset_id + int(float(id))
-                        max_id = max(max_id, id_shifted)
-
-                        if id_shifted not in ids_in_file[name]:
-                            ids_in_file[name].append(id_shifted)
-                        times[name].append(t)
-                        max_id_here = max(max_id_here, id_shifted)
-                        max_time = max(t, max_time)
-                        if id_shifted not in unique_ids:
-                            unique_ids.append(id_shifted)
-                        x.append(t)
-                        y.append(id_shifted)
-
-                # print("max_id_here in %s: %i"%(file_name, max_id_here))
-                labels.append("%s, %s (%i)" % (name, pop, max_id_here - offset_id))
-                offset_id = max_id_here + 1
```

Comment:
I think you've missed this out in your refactoring---for each population, the offset needs to be updated, no?